src/game_pictures.cpp

```diff
@@ -255,56 +349,60 @@ void Game_Picture::Move(const MoveParams& params) {
 	}
 }
 
-void Game_Picture::Erase() {
+void Game_Pictures::Move(int id, const MoveParams& params) {
+	auto& pic = GetPicture(id);
+	pic.Move(params);
+}
+
+void Game_Pictures::Picture::Erase() {
 	request_id = {};
 	data.name.clear();
 	sprite.reset();
-	bitmap.reset();
 }
 
-void Game_Picture::RequestPictureSprite() {
-	const std::string& name = data.name;
+void Game_Pictures::Erase(int id) {
+	auto* pic = GetPicturePtr(id);
+	if (EP_LIKELY(pic)) {
+		pic->Erase();
+	}
+}
+
+void Game_Pictures::RequestPictureSprite(Picture& pic) {
+	const auto& name = pic.data.name;
 	if (name.empty()) return;
 
 	FileRequestAsync* request = AsyncHandler::RequestFile("Picture", name);
 	request->SetGraphicFile(true);
 
-	int pic_id = data.ID;
+	int pic_id = pic.data.ID;
 
 	// FIXME: This lambda is here because it's possible the picture vector can be
 	// resized before the async call of onPictureSpriteReady().
 	// Pictures should be refactored so this ugly hack is not needed.
```

Comment:
Considering that this is now refactored this lambda isn't needed anymore?

Reply:
We need to capture the `pic_id` so yes. It could be done other ways, but there is nothing wrong with the lambda really.

Reply:
Can you remove the FIXME comment at least?

(The other way would be ``request->Bind(&Game_Pictures::OnPictureSpriteReady, this, pic_id)`` as done everywhere else (bet this was added pre-C++11, looks so dated with lambda support))

Reply:
Oops, that FIXME should have been removed. Took it out, thanks!